tools/snapshot/snapshot_builder.cc

```diff
@@ -112,55 +113,44 @@ std::string SnapshotBuilder::Generate(
       creator.SetDefaultContext(Context::New(isolate));
       isolate_data_indexes = main_instance->isolate_data()->Serialize(&creator);
 
-      TryCatch bootstrapCatch(isolate);
-      Local<Context> context = NewContext(isolate);
-      if (bootstrapCatch.HasCaught()) {
-        Local<Object> obj = bootstrapCatch.Exception()->ToObject(context)
-            .ToLocalChecked();
-        Local<Value> stack = obj->Get(
-            context,
-            FIXED_ONE_BYTE_STRING(isolate, "stack")).ToLocalChecked();
-        if (stack->IsUndefined()) {
-          Local<String> str = obj->Get(
-              context,
-              FIXED_ONE_BYTE_STRING(isolate, "name"))
-            .ToLocalChecked()->ToString(context).ToLocalChecked();
-          str = String::Concat(
-            isolate,
-            str,
-            FIXED_ONE_BYTE_STRING(isolate, ": "));
-          stack = String::Concat(
-            isolate,
-            str,
-            obj->Get(
-                context,
-                FIXED_ONE_BYTE_STRING(isolate, "message"))
-              .ToLocalChecked()->ToString(context).ToLocalChecked());
+      // Run the per-contenxt scripts
+      Local<Context> context;
+      {
+        TryCatch bootstrapCatch(isolate);
+        context = NewContext(isolate);
+        if (bootstrapCatch.HasCaught()) {
+          PrintCaughtException(isolate, context, bootstrapCatch);
+          abort();
         }
-        v8::String::Utf8Value utf8_value(isolate, stack);
-        if (*utf8_value != nullptr) {
-          std::string out(*utf8_value, utf8_value.length());
-          fprintf(stderr, "Had Exception: %s\n", out.c_str());
-        } else {
-          fprintf(stderr, "Unknown JS Exception\n");
-        }
-        abort();
       }
       Context::Scope context_scope(context);
 
+      // Create the environment
       env = new Environment(main_instance->isolate_data(),
                             context,
                             args,
                             exec_args,
                             nullptr,
                             node::EnvironmentFlags::kDefaultFlags,
                             {});
-      env->RunBootstrapping().ToLocalChecked();
+      // Run scripts in lib/internal/bootstrap/
+      {
+        TryCatch bootstrapCatch(isolate);
+        v8::MaybeLocal<Value> result = env->RunBootstrapping();
+        if (bootstrapCatch.HasCaught()) {
+          PrintCaughtException(isolate, context, bootstrapCatch);
```

Comment:
```suggestion
          PrintCaughtException(isolate, context, bootstrapCatch);
          abort();
```

Reply:
There is already a `result.ToLocalChecked();` below which should abort the process if there's an exception